transport/whisper/envelopes.go

```diff
@@ -128,12 +132,13 @@ func (m *EnvelopesMonitor) GetState(hash common.Hash) EnvelopeState {
 
 // handleEnvelopeEvents processes whisper envelope events
 func (m *EnvelopesMonitor) handleEnvelopeEvents() {
-	events := make(chan whisper.EnvelopeEvent, 100) // must be buffered to prevent blocking whisper
+	events := make(chan whispertypes.EnvelopeEvent, 100) // must be buffered to prevent blocking whisper
 	sub := m.w.SubscribeEnvelopeEvents(events)
 	defer sub.Unsubscribe()
 	for {
 		select {
 		case <-m.quit:
+			close(events)
```

Comment:
Can we close before unsubscribing? We also don't handle `sub.Error` I think.

Reply:
Do you have a link? I don't know what `sub.Error` you're referring to.

Reply:
`sub` is an interface https://sourcegraph.com/github.com/status-im/whisper@3e303b1d5fc15a373594b56f7ca224ebc5b3c4ca/-/blob/vendor/github.com/ethereum/go-ethereum/event/subscription.go#L41-45.

Reply:
what do we want to do in that scenario? Just return like this?

``` go
case <-sub.Err():
	return
```

Reply:
I guess so. We don't handle errors in any particular way here.

Reply:
But it's also super minor. We can keep it as it is.